math/genvector/inc/Math/GenVector/VectorUtil.h

```diff
@@ -333,6 +338,40 @@ namespace ROOT {
             return vrot;
          }
 
+         /**
+          rotation along a custom axis for a generic vector by an Angle alpha (in rad)
+          returning a new vector.
+          The only pre requisite on the Vector is that it has to implement the X(), Y() and Z()
+          and SetXYZ methods.
+          */
+         template <class Vector>
+         Vector Rotate(const Vector &v, double alpha, const Vector &axis)
+         {
+            if (std::fmod(alpha, 2 * M_PI) == 0.)
+               return v;
+            const double ll = std::sqrt(axis.X() * axis.X() + axis.Y() * axis.Y() + axis.Z() * axis.Z());
+            if (ll == 0.)
+               GenVector::Throw("Axis Vector has zero magnitude");
+            const double sa = std::sin(alpha);
+            const double ca = std::cos(alpha);
+            const double dx = axis.X() / ll;
+            const double dy = axis.Y() / ll;
+            const double dz = axis.Z() / ll;
+            const double rot00 = (1 - ca) * dx * dx + ca, rot01 = (1 - ca) * dx * dy - sa * dz,
+                         rot02 = (1 - ca) * dx * dz + sa * dy, rot10 = (1 - ca) * dy * dx + sa * dz,
+                         rot11 = (1 - ca) * dy * dy + ca, rot12 = (1 - ca) * dy * dz - sa * dx,
+                         rot20 = (1 - ca) * dz * dx - sa * dy, rot21 = (1 - ca) * dz * dy + sa * dx,
+                         rot22 = (1 - ca) * dz * dz + ca;
```

Comment:
```suggestion
            // clang-format off
            const double rot00 = (1 - ca) * dx * dx + ca, rot01 = (1 - ca) * dx * dy - sa * dz, rot02 = (1 - ca) * dx * dz + sa * dy,
            rot10 = (1 - ca) * dy * dx + sa * dz, rot11 = (1 - ca) * dy * dy + ca, rot12 = (1 - ca) * dy * dz - sa * dx,
                         rot20 = (1 - ca) * dz * dx - sa * dy, rot21 = (1 - ca) * dz * dy + sa * dx, rot22 = (1 - ca) * dz * dz + ca;
            // clang-format on
```
Maybe this one looks nicer as a 3x3 grid